app/templates/macros/pagination.html

```diff
@@ -0,0 +1,45 @@
+{%- from "govuk_frontend_jinja/components/pagination/macro.html" import govukPagination -%}
+
+{% macro Pagination(params) %}
+    {% if params.data %}
+        {% set pagination = params.data %}
+        <div class="govuk-grid-row">
+            <div class="govuk-grid-column-full">
+                {% if pagination.pages > 1 %}
+                    {% set prev_link = "?page=" ~ pagination.prev_num if pagination.has_prev else None %}
+                    {% set next_link = "?page=" ~ pagination.next_num if pagination.has_next else None %}
+
+                    {% set pagination_items = namespace(items=[]) %}
```

Comment:
```suggestion
                    {% set pagination_items = [] %}
```

The namespace here feels overkill - can we just use a list and then replace all of `pagination_items.items` with `pagination_items`? Why do we need the extra layer?

Reply:
I've seen the slightly wonky `{% set rows = rows.append() %}` syntax used frequently enough. I agree Jinja doesn't make it easy, but I think it's nicer to do the transformation from python objects to their HTML representation in one place - the jinja templates.

It feels fairly fragmented to have the python code generate some dicts that need to be in some very specific structure for the GOV.UK pagination component. That work happens really far away from where the pagination component is used, so it's harder to understand what is happening and why.

This is why I prefer to do that transformation in the jinja2 templates - because having the related things happen close together is easier to understand.

I agree that we should keep try to minimise the amount of _business_ logic inside jinja2 templates. But this is representation/display logic, and I think templates are very much a good place to have that.